### ISSUE TITLE
I can't allocate adjudicators and venues on Docker Desktop on Mac!

### ISSUE THREAD

**Issue by celikselman** [TabbycatDebate/tabbycat#1388]
Hi, I use the latest Docker Desktop on Mac OSX.
When I click 'Allocate adjudicators' on draw page, I get a blank page.
And I got the same issue with the venues allocation.
What should I do in order to see the allocation tables?

**Comment by philipbelesky** [comment]
Hey, thanks for your report. This sounds like a bug in our docker setup that I'll need a little time to test. One quick thing that might work would be to go to the file `tabbycat/settings/docker.py` and add the following line at the end of the file:

`USE_REDIS_LOCALLY=False`

If you then restart Docker that might help things.

**Comment by celikselman** [comment]
Thanks four your quick reply. I just added the line you mentioned above to the docker.py and restarted Docker, but nothing changed :(

By the way, the downloadable zip file of tabbycat has no .css files. I had to find them one by one from other heroku based tabbycat installions for my osx Docker local installion.

**Comment by philipbelesky** [comment]
So, I think the blank page issue is due to the missing javascript payload. Docker seems to run the compilation process that makes our css/js, but doesn't seem to put them in the right place. It might be a relative path issue or similar; I only dug into it a little bit tonight.

If its viable, you could copy over the /vue/ folder from an existing installation which might help with your issue in the meantime.

**Comment by celikselman** [comment]
Yeah I copied all the 'vue' and sub folders to my local Docker installation directory, even though I was not sure if the all files are necessary :))

I'm looking forward for your solution to the blank page issue :)

By the way, even though it looks like a blank page, when I look at the page source code, I can see almost the same codes (like 90%) of a heroku installation. The difference I see is that the line of '<script async src='https://www.google-analytics.com/analytics.js'></script>' is missing in my Docker installation's allocation pages. Instead of this line, I have a long javascript code part. The rest looks like the same. However, I'm still not sure if this is the reason of the issue, though.

**Comment by philipbelesky** [comment]
Ok, found a degree of progress. It seems like (due to Docker's file system somehow) that the built static assets seem to be wiped in between the container being run. Adding the asset-building step to the image being run seems to address that. 

However, even with the assets working the async stuff that applies the adjudicator allocation is not working. I think this is a slightly different problem to the one you originally reported — there is no blank page, it just doesn't ever load the allocation. I'll have a look into that, but you should at least be able to allocate venues/adjs manually now (if you pull down the hotfix/2.3.3 branch and rebuild the docker image)

**Comment by celikselman** [comment]
YES!!
The hotfix/2.3.3 branch allows to allocate manually as you've said! Thank you very much...

**Comment by philipbelesky** [comment]
Quick update / mental note: the new channels/worker based auto allocation methods don't work without redis being enabled. The ones on legacy pages do. Redis is not enabled for standard local installs and in a Dockerised environment. To address this we could:

- Bundle redis into the local install and Docker install
- Implement an alternate code path that only runs allocators on a worker if `USE_REDIS_LOCALLY` is true

Not sure which path to go down yet. Unifying the dependencies is more attractive at a first glance, but would complicate manual installs (should be seamless in Docker). Either way this is beyond the scope of a hotfix and should be addressed in the version release where the legacy pages are deprecated.
